### PR TITLE
Add support PEP-740 attestations for GitLab CI/CD

### PR DESCRIPTION
**Pull request by facutuesca**
Adds support for uploading PEP-740 attestations generated from a GitLab CI/CD workflow.
The actual verification is done inside `pypi-attestations`, similar to GitHub.

Tested locally with an attestation signed from GitLab CI/CD:
image

Rekor entry: https://search.sigstore.dev/?logIndex=149980684

Part of https://github.com/pypi/warehouse/issues/17001

cc @woodruffw @di

### CODE REVIEW

**Comment by facutuesca** [comment]
TODO: Once merged, I'll open a PR adding documentation on how to generate the attestations from a GitLab workflow

File: warehouse/attestations/services.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def parse_attestations(
         artifact. Attestations are only allowed when uploading via a Trusted
         Publisher, because a Trusted Publisher provides the identity that will be
         used to verify the attestations.
-        Only GitHub Actions Trusted Publishers are supported.
+        Only GitHub Actions and GitLab CI/CD Trusted Publishers are currently supported.
```
Comment (woodruffw) [inline warehouse/attestations/services.py:154]: FWIW, I think we can just remove this line -- this is documented elsewhere and is encoded in `attestation_identity` being present or now anyways.

Comment (facutuesca) [reply warehouse/attestations/services.py]: fixed!

**Review by woodruffw — APPROVED** [review]
LGTM, great work @facutuesca!

**Comment by woodruffw** [comment]
Flagging one thing (could be done now or in a follow-up): the `publisher` macro also needs to be updated:

https://github.com/pypi/warehouse/blob/dde8b3d51ac46840f42c8bca0f2a67b875aa5a65/warehouse/templates/includes/file-details.html#L15-L24

**Comment by facutuesca** [comment]
> Flagging one thing (could be done now or in a follow-up): the `publisher` macro also needs to be updated:
> 
Good catch, fixed!

**Comment by facutuesca** [comment]
Opened PR for the docs here: https://github.com/pypi/warehouse/pull/17133

**Review by di — APPROVED** [review]

**Pull request merged**